File: diet/Omnivore.cpp
/* NIM/Nama  : 13515078 / Veren Iliana K   */
/* File      : Omnivore.cpp                */

#include "Omnivore.h"

Omnivore::Omnivore()
{
	meatRatio = 0.3;
	vegRatio = 0.3;
}

Omnivore::Omnivore(const Omnivore& O)
{
	meatRatio = C.meatRatio;
	vegRatio = C.meat;
}

Omnivore& Omnivore::operator= (const Omnivore& H)
{
	meatRatio = C.meatRatio;
	vegRatio = C.meat;
	return *this;
}

File: Omnivore.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : Omnivore.java
 */

import java.util.*;
import java.lang.*;

public interface Omnivore {
  /**
   * Mengembalikan nilai rasio daging.
   * @return nilai rasio daging
   */
  abstract public double GetMeatRatio();

  /**
   * Mengembalikan nilai rasio sayur.
   * @return nilai rasio sayur
   */
  abstract public double GetVegRatio();
}
